packages/opentelemetry-sdk-trace-base/src/SpanProcessor.ts

```diff
@@ -18,6 +18,12 @@ import { Context } from '@opentelemetry/api';
 import { ReadableSpan } from './export/ReadableSpan';
 import { Span } from './Span';
 
+
+/**
+ * Public interface of Span class
+ */
+export type WriteableSpan = Pick<Span, keyof Span>;
```

Comment:
What about a type union of:

```suggestion
export type WritableSpan = api.Span & ReadableSpan;
```

In this case, the `name` property has to be updated with interface method `api.Span.updateName` instead of write to `WritableSpan.name = 'foobar'` directly -- since property `name` is a writable property on `Span`.

Reply:
There are a few more differences:
* besides `name` also `status` and `endTime` are writable in `Span` class
* `Span` class allows `setAttribute('foo')` whereas API requires a value
* `duration` and `ended` are getters in `Span` class but `readonly` in `ReadableSpan`

I think none of these points is actually blocking an usecase but as sdk-trace-base is GA the question is if this is breaking (and therefore semver-major) or not.
In general I tend to exclude typescript details from semver contract as even updating the typescript version used can result in breaking changes in generated .d.ts files.

From JS point of view nothing changes, it's typescript only.

Reply:
FWIW the `Span` class is part of the public API of `@opentelemetry/sdk-trace-base` therefore I think we have no real reason to create a "reduced" span interface here. I just want to get rid of including the private/protected part here.

Reply:
I don't have a strong preference for this, as the `SpanProcessor.onStart` is already receiving a `Span` as its first parameter, and people can already write to the property `Span.name` directly -- even though it may not be expected. So I'm fine with the status quo.

Reply:
We can't change it anyway without breaking some users, and the spec states the onStart takes a writeable span anyway.